### PR TITLE
Feat: Unsafe actors AND Fate table

### PR DESCRIPTION
**Pull request by daemitus**
This is going to thoroughly break things. What else is a new API level for though?

- Disable "make field readonly" IDE suggestion.
- Moved ActorTable collection implementations into a partial class. It's mostly boilerplate anyways.
- Added ActorTable[indexer] by IntPtr. Solves the issue of people needed to reflect Dalamud to make new actors.
- Made BaseResolver.Dalamud private, like it should be (was internal).
- ClassJob and World Resolvers "Id" field made into a property.
- Exposed the casting actor offsets as properties in Chara
- Name and FCTag are now `SeString`
- Any remaining fields in ClientState are now Properties
- Fixed 2 bugs in MemoryHelper
- Finished making properties out of any remaining fields in all the Job Gauges.
(EDIT)
- Resolvers renamed to NameResolver, moved to ClientState namespace. 
  - This is so as not to collide with Lumina types, and they aren't used by only Actors
- Position3 moved to Game namespace. Not just an Actor thing.
- DataWindow now uses an Enum for the types of data, no more fiddling with ints
- *FEAT* Fate table in the same unsafe way as Actors
- Some fiddling with CorePlugin
- MemoryHelper.ReadSeString that takes a ClientStructs.Utf8String*

### CODE REVIEW

**Comment by daemitus** [comment]
Any field renaming was from Adam's branch

File: Dalamud/Game/ClientState/Actors/ActorTable.cs [hunk 1]
```diff
@@ -56,37 +53,34 @@ public int Length
...
+            }
+        }
 
-                return null;
+        /// <summary>
+        /// Get an actor at the specified address.
+        /// </summary>
+        /// <param name="address">The actor address.</param>
+        /// <returns>An <see cref="Actor"/> at the specified address.</returns>
+        public Actor this[IntPtr address]
```
Comment (goaaats) [inline Dalamud/Game/ClientState/Actors/ActorTable.cs:76]: Are you sure this is not confusing? Technically, you're not indexing anything.

Comment (daemitus) [reply Dalamud/Game/ClientState/Actors/ActorTable.cs:76]: It could be. would you prefer a GetActor  or CreateActor(IntPtr)

Comment (daemitus) [reply Dalamud/Game/ClientState/Actors/ActorTable.cs:76]: Well, we could just expose CreateActorReference too

Comment (goaaats) [reply Dalamud/Game/ClientState/Actors/ActorTable.cs:76]: I just wanted to ask what the problem with making that public was, I think that'd be more sensible

Comment (daemitus) [reply Dalamud/Game/ClientState/Actors/ActorTable.cs]: pushed

**Review by goaaats — APPROVED** [review]

**Review by goaaats — APPROVED** [review]

**Pull request merged**